### DISCUSSION TITLE
How do you handle install-time dependencies?

### DISCUSSION THREAD

**Discussion by scowalt** [ScoopInstaller/Scoop#5299] [Help and Questions]
I want to create a package for [gomerge](https://github.com/Cian911/gomerge).

This is a command line utility built in golang that contains no Windows binaries. I wanted to do what homebrew would do, which is build the binary from source, then put the binary into the user's PATH.

Building `gomerge` would depend on the `go` package _at install-time_. However, the [current docs](https://scoop-docs.vercel.app/docs/concepts/Dependencies.html) are a bit unclear as to how to include an install-time dependency:

> Runtime dependencies are specified with the depends setting in the [app manifest](https://scoop-docs.vercel.app/docs/concepts/App-Manifests.html). Install-time dependencies are detected based on the file extensions of the urls in the app manifest.

I highly doubt that scoop will know to install golang based on my downloading of a `.7z` file of the source code.

For now, I will add a pre-install script that does `scoop install go`. However, this feels a bit hacky. For example, `go` will stay on the user's system if/when they uninstall `gomerge`, which feels like an unintended side effect. Maybe this is what homebrew does?

**Comment by HUMORCE** [comment]
here's what I would do:
```
{
    "version": "3.1.0",
    "description": "A tool which allows you to quickly bulk merge, approve and close several pull requests from your terminal.",
    "homepage": "https://github.com/Cian911/gomerge",
    "license": "MIT",
    "depends": [
        "main/go",
        "main/make"
    ],
    "architecture": {
        "64bit": {
            "url": "https://github.com/Cian911/gomerge/archive/3.1.0.zip",
            "hash": "330f61d03b1a0f934e4e2fe051263a0efc00fa0be517f7f95fcaabbd8fd9cc87"
        }
    },
    "pre_install": [
        "make -C $dir\\gomerge-$version",
        "Move-Item $dir\\gomerge-$version\\bin\\gomerge $dir\\gomerge.exe",
        "Remove-Item $dir\\gomerge-$version -Recurse -ErrorAction SilentlyContinue"
    ],
    "bin": "gomerge.exe"
}

```

> For example, go will stay on the user's system if/when they uninstall gomerge, which feels like an unintended side effect.

In this case: `main/go` and `main/make` is unused packages(orphans), if the user has not installed them before. but Scoop does not marks/tags packages.

I have not tested, but should work:
```
'main/go', 'main/make' | foreach { if (!(scoop info $_).installed) { $deps += $_ + ' ' } }
if ($deps) { scoop install $deps }
...
BUILDING
...
if ($deps) { scoop uninstall $deps }
```

**Reply by rashil2000** [reply]
An alternative would be to `scoop install go` in the pre_install, and `scoop uninstall go` in the post_install.
